server/src/main/java/org/elasticsearch/index/mapper/ObjectMapper.java

```diff
@@ -945,13 +974,16 @@ public void prepare() {
 
             if (ignoredValues != null && ignoredValues.isEmpty() == false) {
                 for (IgnoredSourceFieldMapper.NameValue value : ignoredValues) {
+                    if (filter != null && filter.isPathFiltered(value.name(), XContentDataHelper.isEncodedObject(value.value()))) {
+                        continue;
+                    }
```

Comment:
Don't we have to apply this to the field loaders in the loop right below?

Reply:
I think it's because stored field loaders apply filtering internally.. If so, let's add a comment to the loop below for clarity and awareness.